tests/scheduling_utils.py

```diff
@@ -176,7 +181,8 @@ def check_scheduler_inference_steps(
     random_prompts: bool = False,
     prefix_caching: bool = False,
     seeds: list[int] = None,
-):
+    extra_assert_func: Callable[[EngineCore, dict[str, Any], bool],
+                                Any] = dummy_assert_func):
```

Comment:
it looks like we only use this to pass `verify_block_tables`. Shouldn't we always be validating block tables if one is given in a step ref, or is there a reason that we wouldn't want to?

Reply:
Well, the idea was to move things out of the `check_scheduler_inference_steps()` function which was becoming way to big. (The motivation is similar to the refactoring you did to move the prompt generation out of the function). By passing an extra assert function, we can do test-specific verifications without polluting the function's code. 

Maybe validating block tables is generic enough to be applicable to other tests, but then I think we need another refactoring to make these verifications more modular.